src/CoreFoundation/CFArray.cs

```diff
@@ -119,13 +91,13 @@ public static CFArray FromNativeObjects (params INativeObject[] values)
 
 		public IntPtr GetValue (nint index)
 		{
-			return CFArrayGetValueAtIndex (handle, index);
+			return CFArrayGetValueAtIndex (GetCheckedHandle (), index);
 		}
 
-		public static unsafe IntPtr Create (params IntPtr[] values)
+		internal static unsafe IntPtr Create (params IntPtr[] values)
 		{
-			if (values == null)
-				throw new ArgumentNullException ("values");
+			if (values is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (values));
 			fixed (IntPtr* pv = values) {
```

Comment:
Not a big deal at all, but you can change the P/Invoke to take an `IntPtr[]`, and avoid the fixed/unsafe logic.

Reply:
That's should be measured first [1] (even more since we have some tooling now) and out-of-scope of the PR.

[1] I don't think it matters but, for performance, what I think does not matter 😄